src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IVariableDeclaration.cs

```diff
@@ -665,6 +672,407 @@ void M1()
+                //         /*<bind>*/int[,10] x;/*</bind>*/
+                Diagnostic(ErrorCode.ERR_ValueExpected, "").WithLocation(7, 23),
+                // file.cs(7,28): warning CS0168: The variable 'x' is declared but never used
+                //         /*<bind>*/int[,10] x;/*</bind>*/
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "x").WithArguments("x").WithLocation(7, 28)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclarationSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensionsWithInitializer_VerifyChildren()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        /*<bind>*/int[10] x = { 1 }/*</bind>*/;
+    }
+}
+";
+
+            var compilation = CreateEmptyCompilation(source);
+            (var operation, _) = GetOperationAndSyntaxForTest<VariableDeclarationSyntax>(compilation);
+            var declaration = (IVariableDeclarationOperation)operation;
+            Assert.Equal(2, declaration.Children.Count());
+            Assert.Equal(OperationKind.Literal, declaration.Children.First().Kind);
+            Assert.Equal(OperationKind.VariableDeclarator, declaration.Children.ElementAt(1).Kind);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_VerifyChildren()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        /*<bind>*/int[10] x;/*</bind>*/
+    }
+}
+";
+
+            var compilation = CreateEmptyCompilation(source);
+            (var operation, _) = GetOperationAndSyntaxForTest<VariableDeclarationSyntax>(compilation);
+            var declaration = (IVariableDeclarationOperation)operation;
+            Assert.Equal(2, declaration.Children.Count());
+            Assert.Equal(OperationKind.Literal, declaration.Children.First().Kind);
+            Assert.Equal(OperationKind.VariableDeclarator, declaration.Children.ElementAt(1).Kind);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_VerifyInvalidDimensions()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        int[/*<bind>*/10/*</bind>*/] x;
+    }
+}
+";
+
+            var compilation = CreateEmptyCompilation(source);
+            (var operation, _) = GetOperationAndSyntaxForTest<LiteralExpressionSyntax>(compilation);
+            var declaration = (ILiteralOperation)operation;
+            Assert.True(declaration.ConstantValue.HasValue);
+            Assert.Equal(10, declaration.ConstantValue.Value);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_TestSemanticModel()
+        {
```

Comment:
I imagine this test shouldn't go here, but I wasn't sure where it should go, so I put it here for now.